### ISSUE TITLE
Merge general async guidance into the docs

### ISSUE THREAD

**Issue by timmydo** [dotnet/dotnet-api-docs#7280]
There are some good guides around using async, such as:

https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md

Could I request that these tips be merged into the remarks section of individual class/method/property documents (such as https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource-1?view=net-5.0) as well as maybe having a dedicated section within dotnet or asp.net?

**Comment by timmydo** [comment]
No, I wasn't, but looking at that one, it seems the content doesn't cover the AsyncGuidance.md I linked.

Generally speaking, I don't find the articles on docs.microsoft.com to be as discoverable as the API docs. I'm not sure what I could suggest other than incorporating some of the guidance into the remarks sections.

**Comment by stephentoub** [comment]
David's examples are good, but there's also a lot of opinion mixed in.  I'd be fine if we incorporated some of the more factual aspects that aren't covered in enough detail, or if we wanted to add coverage of the pros and cons to various things, but I don't think the docs should take as hard a stance on some of these as he does.  You mention TaskCompletionSource as an example; there are plenty of situations where passing RunContinuationsAsynchronously, which the write-up says to always do, is a deoptimization and doesn't provide benefit, even though I agree with the general spirit of it being the safer thing to do until you know what you're doing.
